pkg/controllers/dashboardapi/settings/settings.go

```diff
@@ -148,5 +153,67 @@ func (s *settingsProvider) SetAll(settingsMap map[string]settings.Setting) error
 
 	s.fallback = fallback
 
+	if err := s.cleanupUnknownSettings(settingsMap); err != nil {
+		logrus.Errorf("Error cleaning up unknown settings: %v", err)
+	}
+
 	return nil
 }
+
+const unknownSettingLabelKey = "cattle.io/unknown"
+
+// cleanupUnknownSettings lists all the settings in the cluster and cleans up all unknown (e.g. deprecated) settings.
+// At the moment, we just mark such settings with a label.
+// In the future release we'll be removing them.
+func (s *settingsProvider) cleanupUnknownSettings(settingsMap map[string]settings.Setting) error {
+	// The settings cache is not yet available at this point, thus using the client directly.
+	list, err := s.settings.List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, setting := range list.Items {
+		if _, ok := settingsMap[setting.Name]; ok {
+			continue
+		}
+
+		if err := s.markSettingAsUnknown(&setting); err != nil {
+			logrus.Errorf("Error adding label %s to setting %s: %v", unknownSettingLabelKey, setting.Name, err)
+			continue
+		}
+	}
+
+	return nil
+}
+
+// markSettingAsUnknown adds a label to the setting to mark it as unknown.
+func (s *settingsProvider) markSettingAsUnknown(setting *v3.Setting) error {
+	logrus.Warnf("Unknown setting %s", setting.Name)
+
+	var try int
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		defer func() { try++ }()
+
+		var err error
+
+		if try > 0 { // Refetch only if the first attempt to update failed.
```

Comment:
Nit: In reality you are using this as a bool to determine if this has been called a single time, but you are storing/using an int which you update every time. Since this has no meaning past the first iteration, using a bool might make more sense here.